helm/cluster-api-provider-azure/templates/_resource.tpl

```diff
@@ -0,0 +1,27 @@
+{{/* vim: set filetype=mustache: */}}
+{{/*
+Create a name stem for resource names
+When pods for deployments are created they have an additional 16 character
+suffix appended, e.g. "-957c9d6ff-pkzgw". Given that Kubernetes allows 63
+characters for resource names, the stem is truncated to 47 characters to leave
+room for such suffix.
+*/}}
+{{- define "resource.default.name" -}}
+{{- .Release.Name | replace "." "-" | trunc 47 | trimSuffix "-" -}}
+{{- end -}}
+
+{{- define "resource.networkPolicy.name" -}}
+{{- include "resource.default.name" . -}}-network-policy
+{{- end -}}
+
+{{- define "resource.psp.name" -}}
+{{- include "resource.default.name" . -}}-psp
+{{- end -}}
+
+{{- define "resource.pullSecret.name" -}}
+{{- include "resource.default.name" . -}}-pull-secret
+{{- end -}}
+
+{{- define "resource.default.namespace" -}}
+giantswarm
```

Comment:
should we stick to `capz-system` like upstream does?

Reply:
Not sure, it was one of the things we discussed. I'll ask in CAPZ channel. cc @MarcelMue

Reply:
We'll leave `giantswarm` here and I'll open an issue in CAPZ to support configurable namespace, as it should be allowed. Currently in CAPZ it's hardcoded in only in `AzureIdentity` controller/scope.

`clusterctl init` has a parameter `--target-namespace` to specify the namespace where you want to deploy controllers.